plugins/importexport/csv/CSVImportExportPlugin.inc.php

```diff
@@ -80,192 +197,724 @@ function executeCLI($scriptName, &$args) {
+	 *
+	 * @param string $pressPath
+	 */
+	private function getCachedGenreId($pressId) {
+		return $this->_genreIds[$pressId] ?? $this->_genreIds[$pressId] = $this->getGenreIdByPressId($pressId);
+	}
+
+	/**
+	 * Returns a cached Serie ID or create a new one, if it isn't retreived yet.
+	 *
+	 * @param string $seriesPath
+	 * @param int $pressId
+	 */
+	private function getCachedSerieId($seriesPath, $pressId) {
+		$key = "{$seriesPath}_{$pressId}";
+		return $this->_serieIds[$key] ?? $this->_serieIds[$key] = $this->getSerieIdByPathAndPressId($seriesPath, $pressId);
+	}
+
+	/**
+	 * Returns a cached UserGroup ID or create a new one, if it isn't retreived yet.
+	 *
+	 * @param string $pressPath
+	 */
+	private function getCachedUserGroupId($pressId) {
+		return $this->_userGroupIds[$pressId] ?? $this->_userGroupIds[$pressId] = $this->getUserGroupIdByPressId($pressId);
+	}
+
+	/**
+	 * Insert data on the invalid_rows.csv file and increase the failed rows counter
+	 *
+	 * @param SplFileObject $invalidRowsFile
+	 * @param array $fields
+	 * @param int $failedRows
+	 * @param string $reason
+	 */
+	private function processFailedRow($invalidRowsFile, $fields, $reason, $failedRows) {
+		$invalidRowsFile->fputcsv(array_merge(array_pad($fields, $this->_expectedRowSize, null), [$reason]));
+
+		return $failedRows + 1;
+	}
+
+	/**
+	 * Verify if all required fields are present on a CSV row
+	 *
+	 * @param object $row
+	 */
+	private function requiredFieldsPresent($row) {
+		 return !!$row->pressPath
+		 	&& !!$row->authorString
+			&& !!$row->title
+			&& !!$row->abstract
+			&& !!$row->locale
+			&& !!$row->filename;
+	}
+
+	/**
+	 * Retrives a user by username
+	 *
+	 * @param string $username
+	 */
+	private function getUser($username) {
+		/** @var UserDAO $userDao  */
+		$userDao = $this->getCachedDao('UserDAO');
+		return $userDao->getByUsername($username);
+	}
+
+	/**
+	 * Retrieves a Press by path.
+	 *
+	 * @param string $pressPath
+	 * @return Press
+	 */
+	private function getPressByPath($pressPath) {
+		/** @var PressDAO $pressDao */
+		$pressDao = $this->getCachedDao('PressDAO');
+		return $pressDao->getByPath($pressPath);
+	}
+
+	/**
+	 * Retrieves a Genre ID by Press ID. If the Genre doesn't exist, the result
+	 * will be false
+	 *
+	 * @param int $pressId
+	 */
+	private function getGenreIdByPressId($pressId) {
+		/** @var GenreDAO $genreDao */
+		$genreDao = $this->getCachedDao('GenreDAO');
+		$genre = $genreDao->getByKey('MANUSCRIPT', $pressId);
+
+		return !$genre ? false : $genre->getId();
+	}
+
+	/**
+	 * Retrieves a UserGroup ID by press ID. If the UserGroup doesn't exist,
+	 * the result will be false
+	 *
+	 * @param int $pressId
+	 */
+	private function getUserGroupIdByPressId($pressId) {
+		/** @var UserGroupDAO $userGroupDao */
+		$userGroupDao = $this->getCachedDao('UserGroupDAO');
+		$userGroup = $userGroupDao->getDefaultByRoleId($pressId, ROLE_ID_AUTHOR);
+
+		return !$userGroup ? false : $userGroup->getId();
```

Comment:
There are other cases returning `false`, I think `null` fits better.

```suggestion
		return $userGroup ? $userGroup->getId() : null;
```